### PR TITLE
 rust-phonenumber: panic-on-parse

### PR DESCRIPTION
**Pull request by rubdos**
[CVE-2024-39697/GHSA-mjw4-jj88-v687](https://github.com/whisperfish/rust-phonenumber/security/advisories/GHSA-mjw4-jj88-v687).

### CODE REVIEW

File: crates/phonenumber/RUSTSEC-0000-0000.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+id = "RUSTSEC-0000-0000"
+
+package = "phonenumber"
+date = "2024-07-07"
+url = "https://github.com/whisperfish/rust-phonenumber/security/advisories/GHSA-mjw4-jj88-v687"
+references = ["https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2024-00000"]
+categories = ["denial-of-service"]
+cvss = "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:C/C:N/I:N/A:H"
+keywords = ["panic", "untrusted input", "parsing"]
+aliases = ["CVE-2024-00000", "mjw4-jj88-v687"]
```
Comment (amousset) [inline crates/phonenumber/RUSTSEC-0000-0000.md:12]:
```suggestion
aliases = ["CVE-2024-00000", "GHSA-mjw4-jj88-v687"]
```

**Comment by amousset** [comment]
Thanks! Looks good, just a small fix on the aliases.

File: crates/phonenumber/RUSTSEC-0000-0000.md (hunk 1 above)
Comment (rubdos) [reply crates/phonenumber/RUSTSEC-0000-0000.md]: Right, I committed these before the CVE number was known. Force-pushed the correct numbers and the GHSA prefix.

File: crates/phonenumber/RUSTSEC-0000-0000.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+The phonenumber parsing code may panic due to a reachable `assert!` guard on the phonenumber string.
+
+In a typical deployment of rust-phonenumber, this may get triggered by feeding a maliciously crafted phonenumber, e.g. over the network, specifically strings of the form `+dwPAA;phone-context=AA`, where the "number" part potentially parses as a number larger than 2^56.
+
+Since f69abee1/0.3.4/#52.
+
+0.2.x series is not affected.
+
+### Patches
+Patches will be published as version `0.3.6+8.13.36`.
```
Comment (tarcieri) [inline crates/phonenumber/RUSTSEC-0000-0000.md:35]:
```suggestion
Patches have been published as version `0.3.6+8.13.36`.
```

**Comment by rubdos** [comment]
Want me to squash this?

**Pull request merged**